### PR TITLE
Add explanation about the build of our frontend

### PR DESCRIPTION
**Pull request by juliensnz**
**Description**

**Definition Of Done**

| Q                                 | A
| --------------------------------- | ---
| Technical Review and 2 GTM        | Todo
| English Review and 1 GTM          | Todo

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: contribute_to_pim/front.rst [hunk 1]
```diff
@@ -0,0 +1,152 @@
+How to contribute to the frontend part of the application
+=========================================================
+
+Currently, Akeneo PIM is using both assetic and webpack to build the frontend assets: style, javascript, images, translation files and configuration files. Sometimes it can be confusing to know which command to run to get started or update a file.
```
Comment (jjanvier) [inline contribute_to_pim/front.rst]: or *to* update a file

?

File: contribute_to_pim/front.rst [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+The .less files are compiled into a main css file with lessphp.
+
+This command also compiles the old requirejs configuration and translation files into javascript code later injected in the frontend modules.
+
+Run webpack
++++++++++++
+
+The last step is now to run the webpack compilation step to generate the final javascript artefacts.
+
```
Comment (jjanvier) [inline contribute_to_pim/front.rst]: .. code-block:: bash
     yarn run webpack

?

Comment (jjanvier) [reply contribute_to_pim/front.rst]: yarn run webpack-dev

maybe also?

File: contribute_to_pim/front.rst [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    yarn webpack-watch
+
+This will run the initial build of webpack and then recompile each time you modify a javascript file (and reload your browser).
+
+What webpack command to run?
+----------------------------
+
+Akeneo PIM provides three webpack commands to build the javascript artefacts
+
+yarn webpack
```
Comment (jjanvier) [inline contribute_to_pim/front.rst]: yarn run webpack?

File: contribute_to_pim/front.rst [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+++++++++++++
+
+This first command will build the javascript file for production. The javascript will be minified and this process can take a lot of time. It's not advised to use this command in development phase.
+
+yarn webpack-dev
```
Comment (jjanvier) [inline contribute_to_pim/front.rst]: missing run?

File: contribute_to_pim/front.rst [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+++++++++++++++++
+
+This command will build the javascript artefacts in development mode. The size of the generated bundle will be higher and quicker to generate. It's the prefered way to rebuild the frontend after checkouting another branch when you are not actively working on the frontend.
+
+yarn webpack-watch
```
Comment (jjanvier) [inline contribute_to_pim/front.rst]: missing run?

File: contribute_to_pim/front.rst [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+++++++++++++++++++
+
+>This command does exactly what the yarn webpack command does but will not exit at the end of the process.
+Instead, it will wait for modifications and recompile the changed files if needed. It will then reload your browser to see the modifications in the PIM.
+
+Last word about the browser cache
+---------------------------------
+
+We currently don't manage dynamic assets filenames to automatically force browser cache update. So you will need to clear your browser cache when working on the PIM frontend. What we advise to avoid this problem is to disable the browser cache when your debug console is open (this option is available on the main browser in the market).
```
Comment (jjanvier) [inline contribute_to_pim/front.rst]: debug console is open*ed*

File: contribute_to_pim/front.rst [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+Most of the time, when you are contributing to the PIM you do a bit of everything at the same time. Sometimes, it can be hard to keep track of which command to run and when. That's why it can be really handy to define an alias in your terminal to run a compilation of them.
+
+If you touch a configuration file, a .less file or a translation file (anything but a javascript file)
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+run this command:
+
+.. code-block:: bash
+
+    rm -rf ./var/cache/*; rm -rf ./web/js/*; rm -rf ./web/css/*; /usr/local/opt/php\@7.2/bin/php bin/console pim:installer:dump-extensions --env=prod; /usr/local/opt/php\@7.2/bin/php bin/console pim:install:ass --env=prod --symlink;
```
Comment (anaelChardan) [inline contribute_to_pim/front.rst]: The path is for mac only, you can jut put `php` instead ;)

File: contribute_to_pim/front.rst [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+yarn webpack-dev
+++++++++++++++++
+
+This command will build the javascript artefacts in development mode. The size of the generated bundle will be higher and quicker to generate. It's the prefered way to rebuild the frontend after checkouting another branch when you are not actively working on the frontend.
+
+yarn webpack-watch
+++++++++++++++++++
+
+>This command does exactly what the yarn webpack command does but will not exit at the end of the process.
```
Comment (anaelChardan) [inline contribute_to_pim/front.rst]: Extra `>`

File: contribute_to_pim/front.rst (hunk 3 above)
Comment (juliensnz) [reply contribute_to_pim/front.rst]: the run is not needed here

File: contribute_to_pim/front.rst (hunk 7 above)
Comment (juliensnz) [reply contribute_to_pim/front.rst]: ha yeah, good catch

File: contribute_to_pim/front.rst [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+    rm -rf ./var/cache/*; rm -rf ./web/js/*; rm -rf ./web/css/*; /usr/local/opt/php\@7.2/bin/php bin/console pim:installer:dump-extensions --env=prod; /usr/local/opt/php\@7.2/bin/php bin/console pim:install:ass --env=prod --symlink;
+
+We strongly advise you to create an alias
+
+If you only modify a javascript file
+++++++++++++++++++++++++++++++++++++
+
+.. warning::
+
+    If you are only working on javascript files, you don't need to run the previous command. Never
```
Comment (tamarasaurus) [inline contribute_to_pim/front.rst]: Never ?

File: contribute_to_pim/front.rst [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+
+yarn webpack
+++++++++++++
+
+This first command will build the javascript file for production. The javascript will be minified and this process can take a lot of time. It's not advised to use this command in development phase.
+
+yarn webpack-dev
+++++++++++++++++
+
+This command will build the javascript artefacts in development mode. The size of the generated bundle will be higher and quicker to generate. It's the prefered way to rebuild the frontend after checkouting another branch when you are not actively working on the frontend.
```
Comment (tamarasaurus) [inline contribute_to_pim/front.rst]:
```suggestion
This command will build the javascript artefacts in development mode. The size of the generated bundle will be higher and quicker to generate. It's the prefered way to rebuild the frontend after checking out another branch when you are not actively working on the frontend.
```

File: contribute_to_pim/front.rst (hunk 6 above)
Comment (tamarasaurus) [reply contribute_to_pim/front.rst]: we can say chrome, it's not a dirty word right :stuck_out_tongue:

**Pull request merged**